### DISCUSSION TITLE
Issue when Importing pnp/sp

### DISCUSSION THREAD

**Discussion by eldrad95** [microsoft-search/pnp-modern-search#3191] [Q&A]
Hi, 
I'm new to dev custom component for pnp search and I have a little but blocking issue. 
I needed to use pnp/sp in but I have issues when I try. I think I make a mistake somewhere but I don't find answer to my issue. 

when I try to build my solution i have this error messages : 
`Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(12,115): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(12,155): error TS1005: ';' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(12,156): error TS1131: Property or signature expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(12,157): error TS1109: Expression expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(13,5): error TS1128: Declaration or statement expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(13,101): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(13,112): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(13,127): error TS1109: Expression expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(14,5): error TS1128: Declaration or statement expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(14,115): error TS1005: '(' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(15,5): error TS1128: Declaration or statement expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(15,117): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(15,132): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(15,150): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(15,157): error TS1109: Expression expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(16,5): error TS1128: Declaration or statement expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(16,95): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(16,108): error TS1005: ',' expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(16,115): error TS1109: Expression expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(17,5): error TS1128: Declaration or statement expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(17,85): error TS1109: Expression expected.
Error - [tsc] node_modules/@pnp/queryable/queryable.d.ts(18,1): error TS1128: Declaration or statement expected.
Error - 'tsc' sub task errored after 4.76 s `

I think I have a package version issue. 
my package.json file  : 
`{
  "name": "Search-CustomControl",
  "version": "1.0.0",
  "private": false,
  "main": "lib/index.js",
  "scripts": {
    "build": "gulp bundle --ship && gulp package-solution --ship",
    "clean": "gulp clean",
    "test": "gulp test"
  },
  "dependencies": {
    "@microsoft/sp-component-base": "1.15.2",
    "@microsoft/sp-core-library": "1.15.2",
    "@microsoft/sp-http": "1.15.2",
    "@microsoft/sp-lodash-subset": "1.15.2",
    "@microsoft/sp-page-context": "1.15.2",
    "@microsoft/sp-property-pane": "1.15.2",
    "@microsoft/sp-webpart-base": "1.15.2",
    "@pnp/modern-search-extensibility": "1.5.0",
    "@pnp/sp": "^3.15.0",
    "@types/es6-promise": "0.0.33",
    "@types/webpack-env": "1.13.1",
    "@webcomponents/custom-elements": "1.4.0",
    "@webcomponents/webcomponentsjs": "2.4.2",
    "handlebars": "4.7.6",
    "office-ui-fabric-react": "7.156.0",
    "react": "16.9.0",
    "react-dom": "16.9.0",
    "tslib": "^1.11.0",
    "typescript": "~3.3.0"
  },
  "devDependencies": {
    "@microsoft/rush-stack-compiler-3.7": "0.2.8",
    "@microsoft/sp-build-web": "1.15.2",
    "@microsoft/sp-module-interfaces": "1.15.2",
    "@microsoft/sp-tslint-rules": "1.15.2",
    "@microsoft/sp-webpart-workbench": "1.12.1",
    "@types/react": "16.9.36",
    "@types/react-dom": "16.9.8",
    "ajv": "~5.2.2",
    "fancy-log": "1.3.3",
    "gulp": "4.0.2",
    "semver": "6.3.0",
    "tslib": "^1.11.0"
  }
}`
I made the project from the example one. 
without using pnp/ps package it works.
does someone see where is my mistake ?

**Comment by wobba** [comment]
I suggest you ask pnpjs questions over at https://github.com/pnp/pnpjs
